pennylane/ops/functions/generator.py

```diff
@@ -58,32 +58,25 @@ def _generator_prefactor(gen):
       of Pauli words.
     """
 
-    prefactor = 1.0
-
-    gen = qml.simplify(gen) if isinstance(gen, Prod) else gen
```

Comment:
I wanted to get rid of this `simplify` call. Then I realized this whole functions was kinda smelly. We shouldn't differentiate between `LinearCombination` and `Sum`.